### PR TITLE
docs: add React examples for notification

### PR DESCRIPTION
**Pull request by tomivirkki**
Extracted from #2632

### CODE REVIEW

File: frontend/demo/component/notification/react/notification-primary.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        setNotificationOpened(false);
+        notification.removeEventListener('opened-changed', handleOpenChanged);
+      }
+    };
+
+    notification.addEventListener('opened-changed', handleOpenChanged);
+  };
+
+  return (
+    <>
```
Comment (web-padawan) [inline frontend/demo/component/notification/react/notification-primary.tsx:32]: nit: fragment is not needed as `Button` is the only child.

File: frontend/demo/component/notification/react/notification-success.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      if (!e.detail.value) {
+        setNotificationOpened(false);
+        notification.removeEventListener('opened-changed', handleOpenChanged);
+      }
+    };
+    notification.addEventListener('opened-changed', handleOpenChanged);
+  }
+
+  return (
+    <>
```
Comment (web-padawan) [inline frontend/demo/component/notification/react/notification-success.tsx:27]: nit: fragment is not needed as `Button` is the only child.

File: frontend/demo/component/notification/react/notification-rich.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import '@vaadin/icons';
+
+function Example() {
+  function open(event: React.MouseEvent) {
+    const target = event.target as HTMLElement;
+    const notification =
+      document.querySelectorAll<NotificationElement>('h-notification')[
+        Number(target.dataset.which) - 1
+      ];
+    notification?.open();
```
Comment (web-padawan) [inline frontend/demo/component/notification/react/notification-rich.tsx:17]: This example doesn't work: `h-notification` is not the right selector, and using `document` is also incorrect.

File: frontend/demo/component/notification/react/notification-position.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { reactExample } from 'Frontend/demo/react-example'; // hidden-source-line
+import exampleStyles from './notification-position-example-styles'; // hidden-source-line
+import React from 'react';
+import { Button } from '@hilla/react-components/Button.js';
+import { Notification } from '@hilla/react-components/Notification.js';
+import type { NotificationPosition } from '@hilla/react-components/Notification';
```
Comment (web-padawan) [inline frontend/demo/component/notification/react/notification-position.tsx:6]: nit: these two imports could be combined.

File: frontend/demo/component/notification/react/notification-static-helper.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    // tag::snippet[]
+    // Show a simple text-based notification
+    Notification.show('Financial report generated', {
+      position: 'middle',
+    });
+    // end::snippet[]
+  }
+
+  return (
+    <>
```
Comment (web-padawan) [inline frontend/demo/component/notification/react/notification-static-helper.tsx:17]: nit: fragment is not needed as `Button` is the only child.

File: frontend/demo/component/notification/react/notification-rich.tsx (hunk 3 above)
Comment (tomivirkki) [reply frontend/demo/component/notification/react/notification-rich.tsx]: Done

File: frontend/demo/component/notification/react/notification-position.tsx (hunk 4 above)
Comment (tomivirkki) [reply frontend/demo/component/notification/react/notification-position.tsx]: Done

File: frontend/demo/component/notification/react/notification-static-helper.tsx (hunk 5 above)
Comment (tomivirkki) [reply frontend/demo/component/notification/react/notification-static-helper.tsx]: Done

File: frontend/demo/component/notification/react/notification-rich.tsx [hunk 6]
```diff
@@ -72,24 +107,27 @@ function Example() {
...
+          <Button theme="tertiary-inline" onClick={() => close(3)} aria-label="Close">
             <Icon icon="lumo:cross" />
           </Button>
         </HorizontalLayout>
       </Notification>
       {/* end::snippet[] */}
 
       <HorizontalLayout theme="spacing" style={{ justifyContent: 'center' }}>
-        <Button onClick={open} data-which="1" theme="success primary">
+        <Button data-which="1" theme="success primary" onClick={() => open(0)}>
```
Comment (web-padawan) [inline frontend/demo/component/notification/react/notification-rich.tsx:118]: Looks like `data-which` attributes are no longer needed and can be removed now.

Comment (tomivirkki) [reply frontend/demo/component/notification/react/notification-rich.tsx]: Removed

**Pull request merged**